addons/website/models/digest.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class Digest(models.Model):
+    _inherit = 'digest.digest'
+
+    kpi_website_visitor_count = fields.Boolean('Visitors')
+    kpi_website_visitor_count_value = fields.Integer(compute='_compute_kpi_website_visitors_count_value')
+    kpi_website_track_count = fields.Boolean('Tracked page views')  # Non uniq
+    kpi_website_track_count_value = fields.Integer(compute='_compute_kpi_website_track_count_value')
+
+    def _compute_website_visitor_count(self, website_ids, start, end):
+        if len(website_ids) == 0:
+            return 0
+        # The query takes into account that some website_visitor are the same visitor through parent_id
+        self._cr.execute("""
+            SELECT count(distinct COALESCE(v.parent_id, v.id))
+                FROM website_visitor AS v
+                JOIN website_track AS t ON t.visitor_id = v.id AND t.visit_datetime >= %(start)s AND t.visit_datetime < %(end)s
+                WHERE v.website_id IN %(website_ids)s""", {'website_ids': website_ids, 'start': start, 'end': end})
+        return self._cr.fetchone()[0]
+
+    def _compute_website_track_count(self, website_ids, start, end):
+        return self.env['website.track'].search_count(
+            [('visitor_id.website_id', 'in', website_ids),
+             ('visit_datetime', '>=', start),
+             ('visit_datetime', '<', end)]) if len(website_ids) > 0 else 0
+
+    def _compute_by_websites_gen(self, compute_kpi_by_ctx):
+        """ Compute a value for each record using compute_kpi_by_ctx and yield the record and the computed value.
```

Comment:
I have to say that the docstring is not crystal clear :sweat_smile: notably it does not really explain why we go from company based to website based computation (which is the interesting point :) ) .

Reply:
Yes thanks it was not clear. I have reworked that part. It should be simpler to understand now.